### PR TITLE
Issue-1467: Fixes incorrect result of boolean-intersects

### PR DESCRIPTION
**Pull request by chriserickson**
I have a potential fix in place, but can't run figure out how to get tests to run.

When I ran them from inside the package, I get this:
```
localhost:turf-boolean-contains cerickson$ npm run test

> @turf/boolean-contains@6.0.1 pretest /Users/cerickson/source/turf/packages/turf-boolean-contains
> tsc

../turf-boolean-crosses/index.ts(66,41): error TS7006: Parameter 'multiPoint' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(66,53): error TS7006: Parameter 'lineString' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(88,29): error TS7006: Parameter 'lineString1' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(88,42): error TS7006: Parameter 'lineString2' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(106,38): error TS7006: Parameter 'lineString' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(115,34): error TS7006: Parameter 'multiPoint' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(115,46): error TS7006: Parameter 'polygon' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(144,31): error TS7006: Parameter 'lineSegmentStart' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(144,49): error TS7006: Parameter 'lineSegmentEnd' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(144,65): error TS7006: Parameter 'pt' implicitly has an 'any' type.
../turf-boolean-crosses/index.ts(144,69): error TS7006: Parameter 'incEnd' implicitly has an 'any' type.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @turf/boolean-contains@6.0.1 pretest: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @turf/boolean-contains@6.0.1 pretest script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/cerickson/.npm/_logs/2018-08-27T14_48_39_689Z-debug.log


```

### CODE REVIEW

**Comment by chriserickson** [comment]
I have one failing test: 2 equal polygons. Given the above discussion, I'd argue that given 2 equal polygons, one does NOT contain the other. 

@rowanwins: I'd love some thoughts on this. If you think the existing behavior should be maintained (more akin to "covers"), then I can look at what it would take to make it so a crossing line is not allowed, but a line lying along another line is allowed.

**Comment by rowanwins** [comment]
Mmm I think technically the exact same polygons should pass, although I think using [ST_ContainsProperly](https://postgis.net/docs/ST_ContainsProperly.html) it would fail...

> ST_Contains — Returns true if and only if no points of B lie in the exterior of A, and at least one point of the interior of B lies in the interior of A.

Note that in the above the rule is about points on the exterior...

> ST_ContainsProperly — Returns true if B intersects the interior of A but not the boundary (or exterior). A does not contain properly itself, but does contain itself.

Whereas here my understanding is that the borders cannot touch in ContainsProperly....

**Comment by chriserickson** [comment]
@rowanwins : I've updated the branch to pass on the equal polygons case. If it looks good can you merge it? Any idea when an updated version will be available from npm?

File: src/boolean-contains/index.js [hunk 1]
```diff
@@ -227,3 +245,18 @@ export function compareCoords(pair1, pair2) {
 export function getMidpoint(pair1, pair2) {
     return [(pair1[0] + pair2[0]) / 2, (pair1[1] + pair2[1]) / 2];
 }
+
+function extractPolygonBorderAsLineString(feature) {
+    const coords = getGeom(feature).coordinates;
+    const outerRing = coords[0];
+
+    // All polygons should have the same first/last point, but this is not required by GeoJSON, so add it if it is missing.
+    const firstPoint = outerRing[0];
```
Comment (rowanwins) [inline src/boolean-contains/index.js]: According to the [GeoJSON spec](http://geojson.win/#section-3.1.6) (see dotpoint 2) a ring must be closed so I think we could probably skip this step.

File: packages/turf-boolean-intersects/test/false/Polygon/Polygon/issue-1467.geojson [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      }
+    },
+    {
+      "type": "Feature",
+      "properties": {},
+      "geometry": {
+        "type": "Polygon",
+        "coordinates": [
+          [
+            [.1, .1],
```
Comment (rowanwins) [inline packages/turf-boolean-intersects/test/false/Polygon/Polygon/issue-1467.geojson]: It's preferable to have 0 before the decimal point, for example pages like [this](https://github.com/Turfjs/turf/blob/a741b9c16dcdca4588abcb197d56899c8fa5d751/packages/turf-boolean-intersects/test/false/Polygon/Polygon/issue-1467.geojson)  or [this](http://geojson.io) won't be able to render the coords so it's just a extra little pain point when trying to review things quickly

**Comment by rowanwins** [comment]
Hi @chriserickson 

This is looking very close, a couple of minor comments on the latest changes. I've had the review done so I'm getting ready to publish an alpha in the next 24 hours!

File: packages/turf-boolean-intersects/test/false/Polygon/Polygon/issue-1467.geojson (hunk 2 above)
Comment (chriserickson) [reply packages/turf-boolean-intersects/test/false/Polygon/Polygon/issue-1467.geojson]: I don't understand how this file was ever committed. It is in the packages folder for a different module... I've removed it.

Comment (chriserickson) [reply packages/turf-boolean-intersects/test/false/Polygon/Polygon/issue-1467.geojson]: And I didn't know you could preview geojson files like that, cool!

File: src/boolean-contains/index.js (hunk 1 above)
Comment (rowanwins) [reply src/boolean-contains/index.js]: Perhaps an earlier version of the spec...

**Comment by rowanwins** [comment]
Hi @chriserickson 

Shhh don't tell anyone but I've quietly published [this](https://www.npmjs.com/package/turf/v/7.0.0-alpha.1) version, just wanna do a bit more testing before I publicise more widely. I'll double check over this pull and hopefully it should be available in the next few days.

**Comment by chriserickson** [comment]
@rowanwins - any update on this?

**Comment by rowanwins** [comment]
Thanks for this @chriserickson - got a bit tied up trying to fix some module tree-shaking stuff

**Pull request merged**